src/content/docs/guides/Beginners/Your First Shader/0_intro.mdx

```diff
@@ -0,0 +1,57 @@
+---
+title: An Introduction to Shaders
+description: Get started creating your first shader
+sidebar:
+  label: An Introduction to Shaders
+  order: 1
+---
+
+## Foreward
+This tutorial is based on one written by [saada2006](https://saada2006.github.io/minecraft-shader-tutorials/). This newer tutorial has been necessitated by the fact that since the tutorial was written, the general state of development on Minecraft shaders has advanced somewhat, and other developers have identified several areas where this can be improved.
+
+## Prerequisites
+:::caution[Warning]
+**This tutorial is designed for people who already know how to code. If you do not know how to code, you will likely struggle with this tutorial.** A general understanding of how computer graphics work is useful but not necessary. For this reason, at times, you may not be shown all of the code. This is to ensure you are reading enough to understand what you are doing. All of the necessary information is provided for you to achieve the intended end result.
+:::
+
+You will need:
+- A suitable text editor for modifying shader code. One popular choice is [Visual Studio Code](https://code.visualstudio.com/), but any program which can edit text files is sufficient. A simpler but popular editor is [Notepad++](https://notepad-plus-plus.org/).
+- A computer capable of running OpenGL 3.3. If your computer was manufactured in the last 15 years it likely supports OpenGL 3.3. Note that whilst OpenGL works on MacOS, it is deprecated, and some things may not work correctly.
+- A willingness to learn. Shaders are hard, and you can't just pick them up overnight. Do not expect to complete this tutorial and become the next Sonic Ether.
+- An instance of Minecraft with Iris, Optifine, or Oculus installed. Since this tutorial is in the Iris documentation, it is assumed you are using Iris.
+
+It is recommended that you use a Minecraft launcher which supports viewing the logs in realtime. One such choice is [Prism Launcher](https://prismlauncher.org/).
```

Comment:
The vanilla launcher supports realtime logs too btw, it's probably best to at least mention how to enable them for vanilla.